### PR TITLE
Add validate test

### LINKED ISSUE #297: Create a unit test for RateLimitConfig.validate
The `validate` method for config/RateLimitConfig needs a unit test.

### PR DESCRIPTION
**Pull request by Conor-Fleming**
Resolves #297 

Please let me know if there are more cases you'd like tested. The```config.validate``` function seemed fairly straight forward, so I included some test cases that seemed obvious.

### CODE REVIEW

File: config/rate_limit_config_test.go [hunk 1]
```diff
@@ -0,0 +1,48 @@
+package config
+
+import (
+	"errors"
+	"testing"
+	"time"
+)
+
+func TestValidate(t *testing.T) {
```
Comment (prestonvasquez) [inline config/rate_limit_config_test.go:9]:
```suggestion
func TestValidate(t *testing.T) {
t.Parallel()

```

**Review by prestonvasquez — CHANGES_REQUESTED** [review]
@Conor-Fleming Just one small request

File: config/rate_limit_config_test.go (hunk 1 above)
Comment (Conor-Fleming) [reply config/rate_limit_config_test.go:9]: Ah yes, I'll fix that up. Thanks.

**Comment by prestonvasquez** [comment]
@Conor-Fleming Thank you for the contribution!

**Pull request merged**